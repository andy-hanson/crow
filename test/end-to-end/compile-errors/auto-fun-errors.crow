main void()
	()

r record

to json(a r) bare

blah json(a r)

== bool(a r, b nat)

to json(a r, b r)

== bool(a ex, b ex)
ex extern

== bool(a en, b en)
en enum(x)

== comparison(a r, b r)
<=> bool(a r, b r)
to bool(a r)

== bool(a string, b string)
