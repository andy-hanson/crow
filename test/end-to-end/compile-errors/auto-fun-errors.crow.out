exit code: 0
stderr:
test/end-to-end/compile-errors/auto-fun-errors.crow 6:1-6:3 Automatic 'to json' can't be 'bare'.
test/end-to-end/compile-errors/auto-fun-errors.crow 8:1-8:5 Function needs a body. (An automatic function must be named '==', '<=>', or 'to'.)
test/end-to-end/compile-errors/auto-fun-errors.crow 10:1-10:3 '==' must take two parameters of the same type.
test/end-to-end/compile-errors/auto-fun-errors.crow 12:1-12:3 'to' must take a single parameter.
test/end-to-end/compile-errors/auto-fun-errors.crow 14:1-14:3 An automatic function parameter must be a 'record' or 'union' type.
test/end-to-end/compile-errors/auto-fun-errors.crow 17:1-17:3 An automatic function parameter must be a 'record' or 'union' type.
An 'enum' or 'flags' type doesn't need automatic functions; it gets these from 'enum-util' or 'flags-util'.
test/end-to-end/compile-errors/auto-fun-errors.crow 20:1-20:3 '==' must return 'bool'.
test/end-to-end/compile-errors/auto-fun-errors.crow 21:1-21:4 '<=>' must return 'comparison'.
test/end-to-end/compile-errors/auto-fun-errors.crow 22:1-22:3 'to' must return 'json'.
test/end-to-end/compile-errors/auto-fun-errors.crow 24:1-24:3 This function can't be automatic because the type is not fully visible in this context.
